qemu/tests/usb_host.py

```diff
@@ -67,20 +67,24 @@ def usb_dev_unplug():
                           " returns: '%s'" % reply)
         return
 
-    device = params["usb_host_device"]
-    (vendorid, productid) = device.split(":")
+    usb_hostdev = params["usb_devices"].split()[-1]
+    usb_options = params.get("options")
+    if usb_options == "with_vendorid_productid":
+        vendorid = params["vendorid_%s" % usb_hostdev]
+        productid = params["productid_%s" % usb_hostdev]
+        usb_options = "vendorid=%s,productid=%s" % (vendorid, productid)
 
     # compose strings
-    lsusb_cmd = "lsusb -v -d %s" % device
-    monitor_add = "device_add usb-host,bus=usbtest.0,id=usbhostdev"
+    lsusb_cmd = "lsusb -v -d %s:%s" % (vendorid, productid)
+    monitor_add = "device_add usb-host,bus=usbtest.0,id=usb-usbhostdev"
     monitor_add += ",vendorid=0x%s" % vendorid
     monitor_add += ",productid=0x%s" % productid
-    monitor_del = "device_del usbhostdev"
+    monitor_del = "device_del usb-usbhostdev"
     match_add = "New USB device found, "
     match_add += "idVendor=%s, idProduct=%s" % (vendorid, productid)
     match_del = "USB disconnect"
 
-    error_context.context("Check usb device %s on host" % device, logging.info)
+    error_context.context("Check usb device [%s] on host" % usb_options, logging.info)
```

Comment:
What will happen is usb_options is `with_no_hostdev` ?

Reply:
It will lead to usb_negative_test from Line 105 to Line128

Reply:
got it, thanks!